### PR TITLE
[3.3.5] Core/Spells: Fix Glyph of Arcane Shot

### PR DESCRIPTION
**Pull request by Eliminationzx**
Closes https://github.com/TrinityCore/TrinityCore/issues/16500

### CODE REVIEW

File: sql/updates/world/2016_02_10_01_world.sql [hunk 1]
```diff
@@ -0,0 +1,6 @@
+--
+DELETE FROM `spell_proc_event` WHERE `entry`= 56841;
+INSERT INTO `spell_proc_event` VALUES ('56841', '0', '9', '2048', '2048', '2048', '256', '0', '0', '0', '0');
```
Comment (tkrokli) [inline sql/updates/world/2016_02_10_01_world.sql]: Looks OK, but a few notes on file format:
- remove apostrophes ( ' ' ) from numbers, only used for names (text)
- add a LineFeed / NewLine at the end of the file (currently missing)

**Comment by DevRival** [comment]
Work for me!

File: src/server/scripts/Spells/spell_hunter.cpp [hunk 2]
```diff
@@ -946,6 +947,66 @@ class spell_hun_viper_attack_speed : public SpellScriptLoader
...
+        }
+
+        bool CheckProc(ProcEventInfo& eventInfo)
+        {
+            bool found = false;
+            Unit::AuraApplicationMap& Auras = eventInfo.GetProcTarget()->GetAppliedAuras();
+            for (Unit::AuraApplicationMap::const_iterator i = Auras.begin(); i != Auras.end(); ++i)
+            {
+                Aura const* aura = i->second->GetBase();
+                // Search only Serpent Sting, Viper Sting, Scorpid Sting, Wyvern Sting
```
Comment (joschiwald) [inline src/server/scripts/Spells/spell_hunter.cpp]: you have to check spellfamilyname too

flag96 class has a HasFlag() method

File: src/server/scripts/Spells/spell_hunter.cpp [hunk 3]
```diff
@@ -946,6 +947,66 @@ class spell_hun_viper_attack_speed : public SpellScriptLoader
...
+        }
+
+        void HandleProc(AuraEffect const* aurEff, ProcEventInfo& eventInfo)
+        {
+            PreventDefaultAction();
+            SpellInfo const* procSpell = eventInfo.GetSpellInfo();
+            if (!procSpell)
+                return;
+
+            uint32 mana = procSpell->ManaCost + procSpell->ManaCostPercentage *  GetTarget()->GetCreateMana() / 100;
```
Comment (joschiwald) [inline src/server/scripts/Spells/spell_hunter.cpp]: SpellInfo::CalcPowerCost

**Comment by tkrokli** [comment]
Looks like a good PR, since joschiwald is only picking at technical stuff. :+1:
@P-Kito : that codestyle indentation for `public:` etc. seems to be variable in the TC source.

**Comment by P-Kito** [comment]
@tkrokli : @joschiwald always blames me for that coding style :+1:

**Pull request closed without merging**